src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -706,7 +719,10 @@ protected function redirectAfterSave($isNewRecord)
         } elseif ($this->gridField->getList()->byID($this->record->ID)) {
             // Return new view, as we can't do a "virtual redirect" via the CMS Ajax
             // to the same URL (it assumes that its content is already current, and doesn't reload)
-            return $this->edit($controller->getRequest());
+            $message = $controller->getRequest()->getHeader('X-Status') ?? rawurlencode(_t(__CLASS__ . '.SAVEDUP', 'Saved successfully') ?? '');
+            $response = new HTTPResponse($this->edit($controller->getRequest()));
+            $response->addHeader('X-Status', $message);
+            return $response;
```

Comment:
Can you please explain how I can trigger this one?

Reply:
This part is called every time we save or publish or unpublish an existing object.

Since, `edit()` method returns `DBHTMLText` or `ViewableData_Customised` therefore we cannot attach header to it, so we should create new `HTTPResponse` and add header to it. Also we will be more consistent in what we return from `redirectAfterSave` method.